OrbitVulkanLayer/LayerLogic.cpp

```diff
@@ -0,0 +1,167 @@
+  command_buffer_manager_.UnTrackCommandPool(command_pool);
+}
+
+void LayerLogic::PostCallResetCommandPool(VkDevice /*device*/, VkCommandPool /*command_pool*/,
+                                          VkCommandPoolResetFlags /*flags*/) {}
+
+void LayerLogic::PostCallAllocateCommandBuffers(VkDevice device,
+                                                const VkCommandBufferAllocateInfo* allocate_info,
+                                                VkCommandBuffer* command_buffers) {
+  const VkCommandPool& pool = allocate_info->commandPool;
+  const uint32_t cb_count = allocate_info->commandBufferCount;
+  command_buffer_manager_.TrackCommandBuffers(device, pool, command_buffers, cb_count);
+}
+
+void LayerLogic::PostCallFreeCommandBuffers(VkDevice device, VkCommandPool command_pool,
+                                            uint32_t command_buffer_count,
+                                            const VkCommandBuffer* command_buffers) {
+  command_buffer_manager_.UnTrackCommandBuffers(device, command_pool, command_buffers,
+                                                command_buffer_count);
+}
+
+void LayerLogic::PostCallBeginCommandBuffer(VkCommandBuffer /*command_buffer*/,
+                                            const VkCommandBufferBeginInfo* /*begin_info*/) {}
+
+void LayerLogic::PreCallEndCommandBuffer(VkCommandBuffer /*command_buffer*/) {}
+
+void LayerLogic::PreCallResetCommandBuffer(VkCommandBuffer /*command_buffer*/,
+                                           VkCommandBufferResetFlags /*flags*/) {}
+
+void LayerLogic::PostCallQueueSubmit(VkQueue /*queue*/, uint32_t /*submit_count*/,
+                                     const VkSubmitInfo* /*submits*/, VkFence /*fence*/) {}
+
+void LayerLogic::PostCallQueuePresentKHR(VkQueue /*queue*/,
+                                         const VkPresentInfoKHR* /*present_info*/) {}
+void LayerLogic::PostCallGetDeviceQueue(VkDevice device, uint32_t /*queue_family_index*/,
+                                        uint32_t /*queue_index*/, VkQueue* queue) {
+  queue_manager_.TrackQueue(*queue, device);
+}
+void LayerLogic::PostCallGetDeviceQueue2(VkDevice device, const VkDeviceQueueInfo2* /*queue_info*/,
+                                         VkQueue* queue) {
+  queue_manager_.TrackQueue(*queue, device);
+}
+
+void LayerLogic::PostCallEnumeratePhysicalDevices(VkInstance instance,
+                                                  uint32_t* physical_device_count,
+                                                  VkPhysicalDevice* physical_devices) {
+  if (physical_device_count != nullptr && physical_devices != nullptr) {
+    // Map these devices to this instance so that we can map each physical
+    // device back to a dispatch table when handling
+    // EnumerateDeviceExtensionProperties. Note that this is hardly error-proof,
+    // but it's impossible to handle this perfectly since physical devices may
+    // be in use by multiple instances.
```

Comment:
So, what's the consequence of this? It's not error-proof, but also can't be handled perfectly, but it's not clear from the comment this is actually a problem. What happens here in case of an error?

Reply:
I rephrased the comment. Hope it is better. For now, I do not know what to do about this.